### PR TITLE
Upgrade Redis to 5.0.0

### PR DESCRIPTION
**Pull request by tutte**
- Upgrade redis default version to 5.0.0
- Use the variable apply_immediately

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -60,11 +60,16 @@ resource "aws_elasticache_replication_group" "default" {
   parameter_group_name          = "${aws_elasticache_parameter_group.default.name}"
   availability_zones            = ["${slice(var.availability_zones, 0, var.cluster_size)}"]
   automatic_failover_enabled    = "${var.automatic_failover}"
+  apply_immediately             = "${var.apply_immediately}"
```
Comment (Patelvijaykumar) [inline main.tf]: didn't found apply_immediately in the variable file

Comment (tutte) [reply main.tf]: Lines 133-136, was already present but not used.

File: main.tf [hunk 2]
```diff
@@ -122,7 +127,7 @@ module "dns" {
   source    = "git::https://github.com/cloudposse/terraform-aws-route53-cluster-hostname.git?ref=tags/0.2.1"
   enabled   = "${var.enabled == "true" && length(var.zone_id) > 0 ? "true" : "false"}"
   namespace = "${var.namespace}"
-  name      = "${var.name}"
+  name      = "${var.dns_name}"
```
Comment (osterman) [inline main.tf]:
```suggestion
  name      = "${local.dns_name}"
```

File: variables.tf [hunk 3]
```diff
@@ -18,6 +18,10 @@ variable "name" {
   description = "Name"
 }
 
+variable "dns_name" {
+  default = "redis"
+}
```
Comment (aknysh) [inline variables.tf]: add `description`

File: variables.tf [hunk 4]
```diff
@@ -55,8 +59,30 @@ variable "instance_type" {
   description = "Elastic cache instance type"
 }
 
+# format HH24:mm-HH24:mm
+variable "snapshot_window" {
+  default = ""
+}
```
Comment (aknysh) [inline variables.tf]: add `description`

File: variables.tf [hunk 5]
```diff
@@ -55,8 +59,30 @@ variable "instance_type" {
...
+
+variable "snapshot_name" {
+  default = ""
+}
```
Comment (aknysh) [inline variables.tf]: add `description`

File: variables.tf [hunk 6]
```diff
@@ -55,8 +59,30 @@ variable "instance_type" {
...
+
+variable "snapshot_retention_limit" {
+  default = 0
+}
```
Comment (aknysh) [inline variables.tf]: add `description`

File: variables.tf [hunk 7]
```diff
@@ -55,8 +59,30 @@ variable "instance_type" {
...
+
+variable "snapshot_arns" {
+  type    = "list"
+  default = []
```
Comment (aknysh) [inline variables.tf]: add `description`

File: variables.tf [hunk 8]
```diff
@@ -55,8 +59,30 @@ variable "instance_type" {
...
+}
+
 variable "family" {
-  default     = "redis4.0"
+  default = "redis5.0"
```
Comment (aknysh) [inline variables.tf]: add `description`

**Pull request closed without merging**